From 42aceee22a3d9c7d2b9661a5a825bf1fcc47ad7d Mon Sep 17 00:00:00 2001
From: Fei Jiang <fei.jiang@intel.com>
Date: Tue, 12 Jun 2018 10:36:57 +0800
Subject: [PATCH 1099/1676] drm/i915/gvt: remove correlation between plane
 restriction and modeset

If i915.enable_initial_modeset is set, fb dev will be disabled, then there is
no display for Android recovery mode. UOS should has no such limitation,
then removed it for UOS, which don't need set i915.enable_initial_modeset
when plane restriction is enabled.

V2: SOS still need force setting enable_initial_modeset for plane restriction
feature.

Signed-off-by: Fei Jiang <fei.jiang@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.c      | 3 +--
 drivers/gpu/drm/i915/intel_display.c | 3 ++-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index 2d74882..ad94d7a 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -1321,8 +1321,7 @@ static inline int get_max_avail_pipes(struct drm_i915_private *dev_priv)
 	int index = 0;
 
 	if (!intel_vgpu_active(dev_priv) ||
-	    !i915_modparams.avail_planes_per_pipe ||
-	    !i915_modparams.enable_initial_modeset)
+	    !i915_modparams.avail_planes_per_pipe)
 		return INTEL_INFO(dev_priv)->num_pipes;
 
 	for_each_pipe(dev_priv, pipe) {
diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index a6f4a91..cf88a88 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -14860,7 +14860,8 @@ static int intel_sanitize_plane_restriction(struct drm_i915_private *dev_priv)
 {
 	/*plane restriction feature is only for APL for now*/
 	if (!IS_BROXTON(dev_priv) ||
-	    !i915_modparams.enable_initial_modeset) {
+	    (!intel_vgpu_active(dev_priv) &&
+	     !i915_modparams.enable_initial_modeset)) {
 		i915_modparams.avail_planes_per_pipe = 0;
 		DRM_INFO("Turning off Plane Restrictions feature\n");
 	}
-- 
2.7.4

